File: 35_search_insert_position.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        left = 0
        right = len(nums) - 1

        while left <= right:
            mid = (left + right) // 2
            if nums[mid] < target:
                left = mid + 1
            elif nums[mid] == target :
                return mid
            else:
                right = mid - 1
        nums.append(target)
        nums.sort()
        return self.searchInsert(nums, target)

File: 35-search-insert-position.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var searchInsert = function (nums, target) {
  left = 0;
  right = nums.length - 1;

  while (left <= right) {
    mid = Math.floor((left + right) / 2);
    if (nums[mid] < target) {
      left = mid + 1;
    } else if (nums[mid] == target) {
      return mid;
    } else {
      right = mid - 1;
    }
  }
  nums.push(target);
  nums.sort(function (a, b) {
    return a - b;
  });
  return searchInsert(nums, target);
};
